public/docs/ts/latest/cookbook/a11y.jade

```diff
@@ -0,0 +1,1087 @@
+    so. Unless you're building an especially complex interaction flow (e.g. a rich text document
+    editor), and you're familiar with this role and its pitfalls, you should avoid its use.
+
+:marked
+  `HTML 5` provides native `semantic elements` that implicitly carry many of these roles and we recommend that you use these 
+   when possible.
+  
+  Let's have a look at a high level HTML layout for a page using the `HTML 5 Semantic Elements`:
+
+code-example(language="html" escape="html" format="linenums").
+  <header role="banner">
+    <!--Site focused header information.-->
+  </header>
+  <nav role="navigation">
+    <!--Main site navigation-->
+  </nav>
+  <main role="main">
+    <!--Contains the main page content-->
+    <form role="search">
+      <!--Search form-->
+    </form>
+    <form>
+      <!--Normal form-->
+    </form>
+  </main>
+  <aside role="complementary">
+    <!--Supplementary site information-->
+  </aside>
+  <footer role="contentinfo">
+    <!--Site information-->
+  </footer>
+
+:marked
+  For landmark elements, we explicitly assign all roles (including native elements where a default
+  exists) as some browsers do not implement native `semantic elements` correctly.
+
+  When it is totally impossible to use these elements, e.g. when you need to support `HTML 4`,
+  we can still create this structure in our HTML using `ARIA Roles`:
+
+code-example(language="html" escape="html" format="linenums").
+  <div role="banner">
+    <!--Site focused header information.-->
+  </div>
+  <div role="navigation">
+    <!--Main site navigation-->
+  </div>
+  <div role="main">
+    <!--Contains the main page content-->
+    <div role="search">
+      <!--Search form-->
+    </div>
+    <div role="form">
+      <!--Normal form-->
+    </div>
+  </div>
+  <div role="complementary">
+    <!--Supplementary site information-->
+  </div>
+  <div role="contentinfo">
+    <!--Site information-->
+  </div>
+
+.l-sub-section
+  :marked
+    In the previous section we looked at `skiplinks`. These `landmarks` are great skiplink
+    destinations.
+
+:marked
+  ### ARIA Roles: The widget roles
+  
+  The other section of `ARIA Roles` we will briefly look at is `Widget Roles`.
+  
+  The following roles are useful for standalone widgets:
+  - alert
+  - alertdialog
+  - button
+  - checkbox
+  - dialog
+  - gridcell
+  - link
+  - log
+  - marquee
+  - menuitem
+  - menuitemcheckbox
+  - menuitemradio
+  - option
+  - progressbar
+  - radio
+  - scrollbar
+  - slider
+  - spinbutton
+  - status
+  - tab
+  - tabpanel
+  - textbox
+  - timer
+  - tooltip
+  - treeitem
+  
+  There is also a set of roles for `composite widgets`, i.e. widgets built from other widgets.
+  - combobox
+  - grid
+  - listbox
+  - menu
+  - menubar
+  - radiogroup
+  - tablist
+  - tree
+  - treegrid
+
+.l-sub-section
+  :marked
+    The names of these widget roles are self-explanatory. Visit the `W3C` to read more about
```

Comment:
Would it be worth linking to the [design patterns](https://www.w3.org/TR/wai-aria-practices/#aria_ex) too? IMO they have a lot of value but aren't linked very well from the widget roles page.